elastic/kibana
# Adding readonly view for API Keys page

## Summary

Adding a `readonly` view for users with `read_security` cluster privileges

## Release Note

The API Keys screen can be accessed in a Read Only view with the the cluster privilege `read_security`

## Testing Steps
 Login as `elastic` and create a `role` with the `read_security` cluster privilege
<img width="877" alt="Screen Shot 2022-11-09 at 1 03 05 PM" src="https://user-images.githubusercontent.com/21210601/200908865-b11ffe67-106e-45c4-a704-9120b5cc4a38.png">

Create a test user and assign the newly create role, as well as `viewer` and `kibana_admin`
<img width="860" alt="Screen Shot 2022-11-09 at 1 03 48 PM" src="https://user-images.githubusercontent.com/21210601/200909077-710efb9d-4863-4a56-a3c1-65fc979d16b6.png">

Login as the new test user and navigate to Stack Management >  API Keys

Verify there aren't any Create buttons and that the ReadOnly `glasses` icon is in the top right
<img width="1311" alt="Screen Shot 2022-11-09 at 1 04 59 PM" src="https://user-images.githubusercontent.com/21210601/200909224-e291f3cf-39ee-4629-ab75-f355ced80db1.png">

Login as `elastic` and create an API key, remember the name of the key

Go to Dev Tools and use the following script to grant usage of the API key to the test user, use the following block:
```json
POST /_security/api_key/grant
{
  "grant_type": "password",
  "username" : "elastic",  
  "password" : "changeme",  
  "run_as": "test_user",  
  "api_key" : {
    "name": "test-api-key"
  }
}
```

Login as the test user and navigate to the API Keys page, notice the granted API key is displayed, but you are unable to `delete` or `create` new keys
<img width="1058" alt="Screen Shot 2022-11-09 at 1 06 48 PM" src="https://user-images.githubusercontent.com/21210601/200909524-e0a3cc20-9626-4d39-8277-d77c3d795ee0.png">
